Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,442 @@
+    {&PoweredCruiseHandler, &PoweredCruiseDecider},
+    {&CoastingCruiseHandler, &CoastingCruiseDecider},
+    {&BrakeHandler, &BrakeDecider},
+    {&OnePedalDriveHandler, &OnePedalDriveDecider},
+    {&AccelerateCruiseHandler, &AccelerateCruiseDecider}
+};
+
+// Current state
+static TritiumState_t state;
+
+/**
+ * @brief Dumps info to UART during testing
+*/
+#ifdef __TEST_SENDTRITIUM
+void dumpInfo(){
+    printf("-------------------\n\r");
+    printf("State: %d\n\r", state);
+    printf("cruiseEnable: %d\n\r", cruiseEnable);
+    printf("cruiseSet: %d\n\r", cruiseSet);
+    printf("regenToggle: %d\n\r", regenToggle);
+    printf("brakePedalPercent: %d\n\r", brakePedalPercent);
+    printf("accelPedalPercent: %d\n\r", accelPedalPercent);
+    printf("reverseSwitch: %d\n\r", reverseSwitch);
+    printf("forwardSwitch: %d\n\r", forwardSwitch);
+    printf("currentSetpoint: %f\n\r", currentSetpoint);
+    printf("velocitySetpoint: %f\n\r", velocitySetpoint);
+    printf("velocityObserved: %f\n\r", velocityObserved);
+    printf("-------------------\n\r");
+}
+#endif
+
+/**
+ * @brief Reads inputs from the system
+*/
+void readInputs(){
+    #ifndef __TEST_SENDTRITIUM
+    Minion_Error_t err;
+
+    brakePedalPercent = Pedals_Read(BRAKE);
+    accelPedalPercent = Pedals_Read(ACCELERATOR);
+    
+    // UPDATE BUTTONS
+    if(Minion_Read_Input(REGEN_SW, &err) && regenCounter < DEBOUNCE_PERIOD){regenCounter++;}
+    else if(regenCounter > 0){regenCounter--;}
+
+    if(Minion_Read_Input(CRUZ_EN, &err) && cruiseEnableCounter < DEBOUNCE_PERIOD){cruiseEnableCounter++;}
+    else if(cruiseEnableCounter > 0){cruiseEnableCounter--;}
+
+    if(Minion_Read_Input(CRUZ_ST, &err) && cruiseSetCounter < DEBOUNCE_PERIOD){cruiseSetCounter++;}
+    else if(cruiseSetCounter > 0){cruiseSetCounter--;}
+    
+    // UPDATE GEARS
+    bool forwardSwitch = Minion_Read_Input(FOR_SW, &err);
+    bool reverseSwitch = Minion_Read_Input(REV_SW, &err);
+    forwardGear = (forwardSwitch && !reverseSwitch);
+    reverseGear = (!forwardSwitch && reverseSwitch);
+    neutralGear = (!forwardSwitch && !reverseSwitch);
+
+    if(forwardSwitch){
+        cruiseSet = Minion_Read_Input(CRUZ_ST, &err);
+    }
+    else{
+        cruiseEnable = false;
+        cruiseSet = false;
+        regenToggle = false;
+    }
+
+    if(forwardGear ^ reverseGear ^ neutralGear == false)
+    ; // TODO: jump to fault state for incorrect button read
+
+    // DEBOUNCING
+    if(regenCounter == DEBOUNCE_PERIOD){regenButton = true;}
+    else if(regenCounter == 0){regenButton = false;}
+
+    if(cruiseEnableCounter == DEBOUNCE_PERIOD){cruiseEnableButton = true;}
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
```

Comment:
If you choose to follow my suggestions, you can replace the references to `ChargeEnable` with references to `regen_enable`

Reply:
Can we change the name to ChargeEnable in ReadCarCAN? I think it makes more sense as a naming convention so that we don't mix up the one pedal drive mode enable with the charge enable message.

Reply:
Also yes, I will use the getter and update a local variable.

Reply:
Also in terms of naming, I'm going to change the name of regenToggle to onePedalToggle to be more consistent in calling it "One Pedal Drive Mode" so that we don't get confused between the regen enable message from BPS and the one pedal drive enable button.